src/transform.js

```diff
@@ -99,39 +102,56 @@ export default class i18nTransform extends Transform {
   }
 
   _flush(done) {
-    let catalog = {}
-
     if (this.options.sort) {
       this.entries = this.entries.sort((a, b) => a.key.localeCompare(b.key))
     }
 
-    let uniqueCount = this.entries.length
-    for (const entry of this.entries) {
-      const { duplicate, conflict } = dotPathToHash(
-        entry,
-        catalog,
-        {
-          separator: this.options.keySeparator,
-          value: this.options.defaultValue,
-          useKeysAsDefaultValue: this.options.useKeysAsDefaultValue
-        }
-      )
-      if (duplicate) {
-        uniqueCount -= 1
-        if (conflict) {
-          const warning = `Found same keys with different values: ${entry.key}`
-          this.emit('warning', warning)
-          if (this.options.verbose) {
-            warn(warning)
+    for (const locale of this.options.locales) {
+      const catalog = {}
+      const { numbers } = i18next.services.pluralResolver.getRule(locale)
+
+      let countWithPlurals = 0
+      let uniqueCount = this.entries.length
+
+      const transformEntry = (entry, suffix) => {
+        const { duplicate, conflict } = dotPathToHash(
+          entry,
+          catalog,
+          {
+            suffix,
+            separator: this.options.keySeparator,
+            value: this.options.defaultValue,
+            useKeysAsDefaultValue: this.options.useKeysAsDefaultValue
           }
+        )
+
+        if (duplicate) {
+          uniqueCount -= 1
+          if (conflict) {
+            const warning = `Found same keys with different values: ${entry.key}`
+            this.emit('warning', warning)
+            if (this.options.verbose) {
+              warn(warning)
+            }
+          }
+        } else {
+          countWithPlurals += 1
+        }
+      }
+
+      for (const entry of this.entries) {
```

Comment:
The code reads well but I wouldn't mind a comment for the section saying "Generating entries for plurales based on i18next rules" or alike.